pygmo/_py_algorithms.py

```diff
@@ -0,0 +1,131 @@
+import random
+
+try:
```

Comment:
Remove this long try catch. The check the possibility to import scipy has to be done in the __init__. If scipy is not found the error message should be rethrown to the user adding something like:

```
from scipy.optimize import minimize raised an exception, are you sure scipy is installed?
```

Reply:
What is the best way to have an optional dependency to scipy? Check in _init_.py and print a warning if it is not successful?

Reply:
indeed, in the ```__init__``` try to import it in a try catch ImportError, and add to the import error message something specific like i wrote above.

Reply:
I've now replaced the long try/catch and added a check in __init__. I don't see what use it is to add a message to the import error before rethrowing it, though. Isn't it pretty clear what an ImportError means? Also, should the dependency mandatory or optional?

Reply:
The dependency on scipy is optional. The user still sees the algorithm in the list but when instantiating it will be signalled the error.

Ok, no need to add text to the import error.

Reply:
Moved the try/catch into __init__.